presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -122,13 +122,15 @@ public OrcReader(OrcDataSource orcDataSource, MetadataReader metadataReader)
 
         // read metadata
         Slice metadataSlice = completeFooterSlice.slice(0, metadataSize);
-        InputStream metadataInputStream = new OrcInputStream(orcDataSource.toString(), metadataSlice.getInput(), compressionKind, bufferSize);
-        this.metadata = metadataReader.readMetadata(metadataInputStream);
+        try (InputStream metadataInputStream = new OrcInputStream(orcDataSource.toString(), metadataSlice.getInput(), compressionKind, bufferSize, new AggregatedMemoryContext())) {
```

Comment:
You have a lot of cleanups in this commit. I would move things like this to a separate commit